pkg/registry/query.go

```diff
@@ -392,24 +385,254 @@ func channelEntriesThatReplace(b model.Bundle, name string) []*ChannelEntry {
 	return entries
 }
 
-func channelEntriesForBundle(b model.Bundle, ignoreChannel bool) []*ChannelEntry {
+func (q Querier) channelEntriesForBundle(b cBundle, ignoreChannel bool) []*ChannelEntry {
 	entries := []*ChannelEntry{{
-		PackageName: b.Package.Name,
-		ChannelName: b.Channel.Name,
+		PackageName: b.Package,
+		ChannelName: b.Channel,
 		BundleName:  b.Name,
 		Replaces:    b.Replaces,
 	}}
 	for _, s := range b.Skips {
 		// Ignore skips that duplicate b.Replaces. Also, only add it if its
 		// in the same channel as b (or we're ignoring channel presence).
-		if _, inChannel := b.Channel.Bundles[s]; s != b.Replaces && (ignoreChannel || inChannel) {
+		if _, inChannel := q.pkgs[b.Package].Channels[b.Channel].Bundles[s]; s != b.Replaces && (ignoreChannel || inChannel) {
 			entries = append(entries, &ChannelEntry{
-				PackageName: b.Package.Name,
-				ChannelName: b.Channel.Name,
+				PackageName: b.Package,
+				ChannelName: b.Channel,
 				BundleName:  b.Name,
 				Replaces:    s,
 			})
 		}
 	}
 	return entries
 }
+
+type cache struct {
+	digest     string
+	baseDir    string
+	persist    bool
+	pkgs       map[string]cPkg
+	apiBundles map[apiBundleKey]string
+}
+
+func newCache(baseDir string, getDigest func() (string, error), getModel func() (model.Model, error)) (*cache, error) {
```

Comment:
The idea behind the model-only based querier was originally: "we don't care what the source format is, we can always serve the grpc API if you can convert it to the model first"

I'm not totally ready to give up on that idea, but once sqlite support disappears, the only format that even exists is FBC.

Reply:
I'm definitely not arguing about it right now.  

My only concern is that it may be difficult to recognize this as a dead path with the sunset of SQLite indices.  I figured someone will do a 'grep -i sqlite' over the code body, and it would stand out if there were a comment.

Reply:
`NewQuerier` isn't actually used in any of the sqlite flows as far as I know. At this point, it is only used in tests, where its much easier and compile-time safe to create a querier from a `model.Model` than it is to build an fs.FS containing FBC.

With that in mind, I'm guessing we'll keep NewQuerier around even after sqlite support is completely removed.